docs/ios/app-fundamentals/file-compression.md

```diff
@@ -0,0 +1,89 @@
+the dependencies as well as ensuring that the compresison operations will be supported on all platforms as long as they meet the minimun OS requirements.
+
+## General file decompression
+
+As previously mentioned, the Compression Framework has been binded following the stream API. This means that in order to compress data, the developer can use
+the normal patterns used in other IO APIs within .Net. The following sample shows how to decompress data with the Compression Framework which is very similar
+to the API found in the System.IO.Compression.DeflateStream API.
+
+```csharp
+// sample zlib data 
+static byte [] compressed_data = { 0xf3, 0x48, 0xcd, 0xc9, 0xc9, 0xe7, 0x02, 0x00 };
+
+using (var backing = new MemoryStream (compressed_data)) // compress data to read
+using (var decompressing = new CompressionStream (backing, CompressionMode.Decompress, CompressionAlgorithm.Zlib)) // create decompressin stream with the correct algorithm
+using (var reader = new StreamReader (decompressing))
+{
+    // perform the required stream operations
+    Console.WriteLine (reader.ReadLine ());
+}
+```
+
+As with other System.IO.Streams, the CompressionStream implementes the IDisposable interface, and therefore developers should ensure that resources are 
+freed once they are not longed required.
+
+## General file compression
+
+The Compression API also allows to compress data following the same API. Data can be compressed using one of the provided algorithms stated in the CompressionAlgorithm enumerator.
+
+```csharp
+// sample method that copies the data from the source stream to the destination stream
+static void CopyStream (Stream src, Stream dest)
+{
+    byte[] array = new byte[1024];
+    int bytes_read;
+    bytes_read = src.Read (array, 0, 1024);
+    while (bytes_read != 0) {
+        dest.Write (array, 0, bytes_read);
+        bytes_read = src.Read (array, 0, 1024);
+    }
+}
+
+static void CompressExample ()
+{
+    // create some sample data to compress
+    byte [] data = new byte[100000];
+    for (int i = 0; i < 100000; i++) {
+        data[i] = (byte) i;
+    }
+
+    using (var dataStream = new MemoryStream (data)) // stream that contains the data to compress
+    using (var backing = new MemoryStream ()) // stream in which the compress data will be written
+    using (var compressing = new CompressionStream (backing, CompressionMode.Compress, CompressionAlgorithm.Zlib, true))
+    {
+        // copy the data to the compressing stream
+        CopyStream (dataStream, compressing);
+        // at this point, the CompressionStream contains all the data from the dataStream but compressed
+        // using the zlib algorithm
+    }
+```
+
+## Async support
+
+The CompressionStream supports all the async operations that are supported by the System.IO.DeglateStream, this means that developers can use
```

Comment:
should be `CompressionStream `

Reply:
Where? The class to document is CompressionStream, which has the same implementation as the DeflateStream (The name might be confusing, we use Compression because the framework is the Compression Framework :/ )

Reply:
I was referring the usage of code highlight. ``

Reply:
@jzeferino ah! the comments in github tricked me